nbconvert/preprocessors/tests/test_execute.py

```diff
@@ -238,18 +250,66 @@ def assert_notebooks_equal(expected, actual):
         ("Unicode.ipynb", dict(kernel_name="python")),
         ("UnicodePy3.ipynb", dict(kernel_name="python")),
         ("update-display-id.ipynb", dict(kernel_name="python")),
+        ("Check History in Memory.ipynb", dict(kernel_name="python")),
     ]
 )
 def test_run_all_notebooks(input_name, opts):
     """Runs a series of test notebooks and compares them to their actual output"""
     input_file = os.path.join(current_dir, 'files', input_name)
-    res = ResourcesDict()
-    res['metadata'] = ResourcesDict()
-    res['metadata']['path'] = os.path.join(current_dir, 'files')
-    input_nb, output_nb = run_notebook(input_file, opts, res)
+    input_nb, output_nb = run_notebook(input_file, opts, notebook_resources())
     assert_notebooks_equal(input_nb, output_nb)
 
 
+def label_parallel_notebook(nb, label):
```

Comment:
I think this would more idiomatically be treated as a custom preprocessor. You would pass in via resources the label value and then overwrite the `preprocess` method:

https://github.com/jupyter/nbconvert/blob/8af6aac0fe9f9096c1886a90449cbd66e6a9f835/nbconvert/preprocessors/base.py#L51-L70

Reply:
That makes sense – do you think that is worthwhile for this set of tests?

I was hesitant to re-implement papermill's templating here, but happy to do so if that makes it more maintainable.

Reply:
I share that hesitation just for testing purposes.